percona/pmm/pmm.go

```diff
@@ -0,0 +1,190 @@
+package pmm
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	"github.com/percona/percona-postgresql-operator/pkg/apis/pg.percona.com/v2beta1"
+)
+
+func SidecarContainer(pgc *v2beta1.PerconaPGCluster) corev1.Container {
+	ports := []corev1.ContainerPort{{ContainerPort: 7777}}
+	for port := 30100; port <= 30105; port++ {
+		ports = append(ports, corev1.ContainerPort{ContainerPort: int32(port)})
+	}
+
+	pmmSpec := pgc.Spec.PMM
+
+	return corev1.Container{
+		Name:            "pmm-client",
+		Image:           pmmSpec.Image,
+		ImagePullPolicy: pmmSpec.ImagePullPolicy,
+		SecurityContext: pmmSpec.ContainerSecurityContext,
+		Ports:           ports,
+		Resources:       pmmSpec.Resources,
+		LivenessProbe: &corev1.Probe{
+			ProbeHandler: corev1.ProbeHandler{
+				HTTPGet: &corev1.HTTPGetAction{
+					Path:   "/local/Status",
+					Port:   intstr.FromInt(7777),
+					Scheme: corev1.URISchemeHTTP,
+				},
+			},
+			FailureThreshold:    3,
+			InitialDelaySeconds: 60,
+			PeriodSeconds:       10,
+			SuccessThreshold:    1,
+			TimeoutSeconds:      5,
+		},
+		Lifecycle: &corev1.Lifecycle{
+			PreStop: &corev1.LifecycleHandler{
+				Exec: &corev1.ExecAction{
+					Command: []string{
+						"bash",
+						"-c",
+						"pmm-admin inventory remove node --force $(pmm-admin status --json | python -c \"import sys, json; print(json.load(sys.stdin)['pmm_agent_status']['node_id'])\")",
+					},
+				},
+			},
+		},
+		Env: []corev1.EnvVar{
+			{
+				Name: "POD_NAME",
+				ValueFrom: &corev1.EnvVarSource{
+					FieldRef: &corev1.ObjectFieldSelector{
+						APIVersion: "v1",
+						FieldPath:  "metadata.name",
+					},
+				},
+			},
+			{
+				Name: "POD_NAMESPACE",
+				ValueFrom: &corev1.EnvVarSource{
+					FieldRef: &corev1.ObjectFieldSelector{
+						APIVersion: "v1",
+						FieldPath:  "metadata.namespace",
+					},
+				},
+			},
+			{
+				Name:  "PMM_USER",
+				Value: "api_key",
+			},
+			{
+				Name:  "PMM_SERVER",
+				Value: pmmSpec.ServerHost,
+			},
+			{
+				Name:  "PMM_AGENT_SERVER_ADDRESS",
+				Value: pmmSpec.ServerHost,
+			},
+			{
+				Name:  "PMM_AGENT_SERVER_USERNAME",
+				Value: "api_key",
+			},
+			{
+				Name: "PMM_AGENT_SERVER_PASSWORD",
+				ValueFrom: &corev1.EnvVarSource{
+					SecretKeyRef: &corev1.SecretKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: pmmSpec.Secret,
+						},
+						Key: "PMM_API_KEY",
```

Comment:
`PMM_SERVER_KEY` we need to be consistent with all other our operators

Reply:
https://github.com/percona/percona-postgresql-operator/commit/422611dd74c87af724c7bf38f628e6c5286f291a